tests/test_schema.py

```diff
@@ -2097,3 +2097,19 @@ class Model(BaseModel):
         'properties': {'a': {'title': 'A', 'type': 'string'}},
         'required': ['a'],
     }
+
+
+def test_multiple_nested_model_declarations_do_not_raise_error():
+    class ModelOne(BaseModel):
+        class NestedModel(BaseModel):
+            a: float
+
+        nested: NestedModel
+
+    class ModelTwo(BaseModel):
+        class NestedModel(BaseModel):
+            b: float
+
+        nested: NestedModel
+
+    schema([ModelOne, ModelTwo])
```

Comment:
We need to check against an expected schema (or at least the names part of the schema), not just confirm no error is raised.

We need a test which locks the exact model name behaviour.

Reply:
Something like the below?

```python
schema_definitions = schema([ModelOne, ModelTwo])['definitions']
assert len(schema_definitions) == 4

# All expected model names to be generated
model_names = (
    'tests__test_schema__test_multiple_nested_model_declarations_do_not_raise_error__<locals>__ModelOne__NestedModel',
    'ModelOne',
    'tests__test_schema__test_multiple_nested_model_declarations_do_not_raise_error__<locals>__ModelTwo__NestedModel',
    'ModelTwo'
)
assert all(name in schema_definitions for name in model_names)
```

Reply:
Something like this yes. Probably better to use `create_module` fixture

Reply:
Thanks,  very helpful stuff.